src/ui/Controls/NikseComboBox.cs

```diff
@@ -1029,8 +1029,11 @@ private void EnsureListViewInitialized()
                     return;
                 }
 
-                var cachedCount = _listView.Items.Count;
-                for (var i = 0; i < cachedCount; i++)
+                var topItem = _listView.TopItem;
+                // calculate visible range
+                var count = Math.Min(topItem.Index + _listView.Bounds.Height / topItem.Bounds.Height, _listView.Items.Count);
+                
+                for (var i = topItem.Index; i < count; i++)
```

Comment:
For example in subtitle format combobox, instead of looping through 300+ item we can just do it to 14/15 (visible items)